### PR TITLE
Clarifying statement regarding deposit w/promo calculation

### PR DESCRIPTION
**Pull request by eballantine**
Jorge asked me today if a deposit is calculated on price with/without promotion applied. I had asked this previously of Ian. Adding to docs for no further confusion!

### CODE REVIEW

File: docs/reference/objects/product/variant/index.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ When using a `variant` object you have access to the following attributes:<br>
 # variant.deposit
 
 Returns the [deposit]({% link docs/reference/objects/product/deposit.md %}) of this variant if paying with deposit is enabled, `nil` otherwise.
+If the variant has an active [promotion]({% link docs/reference/objects/product/promotion.md %}) the deposit rate is applied to the promotional price.
```
Comment (geoffreymm) [inline docs/reference/objects/product/variant/index.md:16]: @eballantine reading this actually, I'm a bit puzzled. I don't think the deposit makes any assumption on where it's being applied right? 

I think this statement makes sense but I'm not sure it should be here?
I'd like to see how are you using this to understand a bit better where we might put this information.

**Review by geoffreymm — COMMENTED** [review]
Nice one on clarifying this type of things on the docs 💪 

I'm just slightly confused on what are we trying to achieve here, the deposit drop just returns rates and not really applying or doing any calculation. 

I assume we use the money filter to do the `apply_promotion` and get how much a given variant cost, then we calculate how much is the initial deposit amount right?

File: docs/reference/objects/product/variant/index.md (hunk 1 above)
Comment (eballantine) [reply docs/reference/objects/product/variant/index.md]: @geoffreymm ahh yes, you're right to be puzzled, the guidance to a theme developer is that it '_should be_ applied' to promotional price. 
My '_is_ applied' is referring to deposit amount that the customer will be asked to pay at checkout. 
I've updated to reflect, is that clear now or should I maybe include an example/s?

Comment (geoffreymm) [reply docs/reference/objects/product/variant/index.md]: @eballantine lovely!! I mean, as a courtesy for the next theme developer, a lil example wouldn't hurt, what do you think?

Comment (eballantine) [reply docs/reference/objects/product/variant/index.md]: @geoffreymm Only polite, go on then.

**Review by geoffreymm — APPROVED** [review]
Sounds good!! 

I think it's good to go. (Have just left 2 comments that weren't addressed, what do you think?)

**Pull request merged**